rocks/RocksStorage.java

```diff
@@ -297,25 +311,27 @@ public Schema(RocksDatabase database, RocksTransaction transaction) {
         public void put(byte[] key, byte[] value) {
             assert isOpen() && !isReadOnly;
             try {
-                if (transaction.isOpen()) readWriteLock.writeLock().lock();
+                readWriteLock.writeLock().lock();
+                if (!isOpen()) throw GraknException.of(RESOURCE_CLOSED);
                 storageTransaction.put(key, value);
             } catch (RocksDBException e) {
                 throw exception(e);
             } finally {
-                if (transaction.isOpen()) readWriteLock.writeLock().unlock();
+                readWriteLock.writeLock().unlock();
             }
         }
 
         @Override
         public void putUntracked(byte[] key, byte[] value) {
             assert isOpen() && !isReadOnly;
             try {
-                if (transaction.isOpen()) readWriteLock.writeLock().lock();
```

Comment:
why did we hav this check?

Reply:
We should use a local boolean to remember to unlock, not remove it altogether